packages/typespec-client-generator-core/src/methods.ts

```diff
@@ -727,6 +727,13 @@ export function createSdkMethods<TServiceOperation extends SdkServiceOperation>(
     const operationGroupClient = diagnostics.pipe(
       createSdkClientType<TServiceOperation>(context, operationGroup, sdkClientType),
     );
+    // skip if there is no operation or sub-client
+    if (
+      operationGroupClient.methods.length === 0 &&
+      (!operationGroupClient.children || operationGroupClient.children.length === 0)
+    ) {
+      continue;
+    }
```

Comment:
you should add this logic to operation group related function. this place is just a consumer.

Reply:
This function is invoked recursively, this is the place the cut the unnecessary recursion properly.

Reply:
And we don't have methods or children/sub-client from an operation group, we need to do this after L728 where client is constructed.

Reply:
implemented in `listOperationGroups` instead